### PR TITLE
Some little enhancements...

### PR DESCRIPTION
**Pull request by dkm**
GPIO change needed for using smartleds crate with spi backend.
Basic support for timers

### CODE REVIEW

**Comment by dkm** [comment]
I'll fix the formatting error later today :)

File: tm4c123x-hal/src/lib.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@
 //!
 //! [`f3`]: https://docs.rs/f3/~0.5.1
 
-#![deny(missing_docs)]
```
Comment (thejpster) [inline tm4c123x-hal/src/lib.rs]: We should probably turn this back on.

File: tm4c123x-hal/src/timer.rs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    TIMER5: (timer5, Timer5),
+
+    WTIMER0: (wtimer0, WideTimer0),
+    WTIMER1: (wtimer1, WideTimer1),
+    WTIMER2: (wtimer2, WideTimer2),
+    WTIMER3: (wtimer3, WideTimer3),
+    WTIMER4: (wtimer4, WideTimer4),
+    WTIMER5: (wtimer5, WideTimer5),
+
+    // TIM3: (tim3, tim3en, tim3rst),
```
Comment (thejpster) [inline tm4c123x-hal/src/timer.rs]: Can probably delete these lines?

**Comment by thejpster** [comment]
A couple of minor nits, but otherwise this looks great. Thanks!

File: tm4c123x-hal/src/lib.rs (hunk 1 above)
Comment (dkm) [reply tm4c123x-hal/src/lib.rs]: Ooops !

File: tm4c123x-hal/src/timer.rs (hunk 2 above)
Comment (dkm) [reply tm4c123x-hal/src/timer.rs]: Ooops^2 ! Sorry for letting these 1 stupid mistakes in the changes...

**Pull request merged**